### DISCUSSION TITLE
custom onChange handling when using mantine-form

### DISCUSSION THREAD

**Discussion by Huiet** [mantinedev/mantine#2702] [Q&A] [answered]
peep the stackoverflow post I made on this:
https://stackoverflow.com/questions/74062183/using-a-custom-onchange-event-on-a-mantine-form-input

tl;dr:
what's the best way to deal with custom onchange handling when using mantine forms.
The spread syntax is what ties things into the form context (I assume), so it's necessary to call the onchange returned from there.
am I doing dangerous things with my approach??

**Comment by rtivital** [comment, marked as answer]
There is no magic in `form.getInputProps`, it is just a shorthand for `value`, `onChange`, `onFocus` and `onBlur` props. You can set these props on your side if you need more control:

```tsx
<TextInput value={form.values.input} onChange={event => form.setFieldValue('input', event.target.value)} />
```

**Reply by Huiet** [reply]
that is super helpful, thank you.

I hadn't realized that was all the onChange did, I had assumed there was more magic under the hood.

Finding the use of this library to be a delight by the way, nice work 😎

**Comment by rajivchaulagain** [comment]
@rtivital what about the error message should we provide then implicitly ??

**Reply by martitv** [reply]
No, `error` also comes from `getInputProps`. If you want to add props to `getInputProps` you can use the `enhanceGetInputProps` option in useForm-hook
